app/depict/src/main/java/org/openscience/cdk/depict/DepictionGenerator.java

```diff
@@ -0,0 +1,667 @@
+
+import com.google.common.collect.FluentIterable;
+import org.openscience.cdk.CDKConstants;
+import org.openscience.cdk.exception.CDKException;
+import org.openscience.cdk.geometry.GeometryUtil;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IBond;
+import org.openscience.cdk.interfaces.IChemObject;
+import org.openscience.cdk.interfaces.IReaction;
+import org.openscience.cdk.layout.StructureDiagramGenerator;
+import org.openscience.cdk.renderer.RendererModel;
+import org.openscience.cdk.renderer.SymbolVisibility;
+import org.openscience.cdk.renderer.color.CDK2DAtomColors;
+import org.openscience.cdk.renderer.elements.Bounds;
+import org.openscience.cdk.renderer.elements.ElementGroup;
+import org.openscience.cdk.renderer.elements.IRenderingElement;
+import org.openscience.cdk.renderer.generators.BasicSceneGenerator;
+import org.openscience.cdk.renderer.generators.IGenerator;
+import org.openscience.cdk.renderer.generators.IGeneratorParameter;
+import org.openscience.cdk.renderer.generators.standard.SelectionVisibility;
+import org.openscience.cdk.renderer.generators.standard.StandardGenerator;
+
+import javax.vecmath.Point2d;
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A high-level API for depicting molecules and reactions.
+ * <p/>
+ * <h4>General Usage</h4>
+ * Create a generator and reuse it for multiple depictions. Configure how
+ * the depiction will look using {@code with...()} methods.
+ * <pre>{@code
+ * DepictionGenerator dg = new DepictionGenerator().withSize(512, 512)
+ *                                                 .withAtomColors();
+ * for (IAtomContainer mol : mols) {
+ *   dg.depict(mol).writeTo("~/mol.png");
+ * }
+ * }</pre>
+ * <p/>
+ * <h4>One Line Quick Use</h4>
+ * For simplifed use we can create a generator and use it once for a single depiction.
+ * <pre>{@code
+ * new DepictionGenerator().depict(mol)
+ *                         .writeTo("~/mol.png");
+ * }</pre>
+ * The intermediate {@link Depiction} object can write to many different formats
+ * through a variety of API calls.
+ * <pre>{@code
+ * Depiction depiction = new DepictionGenerator().depict(mol);
+ * <p/>
+ * // quick use, format determined by name by path
+ * depiction.writeTo("~/mol.png");
+ * depiction.writeTo("~/mol.svg");
+ * depiction.writeTo("~/mol.pdf");
+ * depiction.writeTo("~/mol.jpg");
+ * <p/>
+ * // manually specify the format
+ * depiction.writeTo(Depiction.SVG_FMT, "~/mol");
+ * <p/>
+ * // convert to a Java buffered image
+ * BufferedImage img = depiction.toImg();
+ * <p/>
+ * // get the SVG XML string
+ * String svg = depiction.toSvgStr();
+ * }</pre>
+ *
+ * @author John may
+ */
+public final class DepictionGenerator {
```

Comment:
Great work! I think putting this in a separate modules makes a lot of sense, while at the same time not too application level, as I can easily so how full application GUIs use this...